error-prone-contrib/src/main/java/tech/picnic/errorprone/bugpatterns/util/SourceCode.java

```diff
@@ -1,16 +1,53 @@
 package tech.picnic.errorprone.bugpatterns.util;
 
 import com.google.errorprone.VisitorState;
+import com.sun.source.tree.ExpressionTree;
 import com.sun.source.tree.Tree;
+import com.sun.tools.javac.jvm.Target;
 
 /**
  * A collection of Error Prone utility methods for dealing with the source code representation of
  * AST nodes.
  */
 // XXX: Can we locate this code in a better place? Maybe contribute it upstream?
 public final class SourceCode {
+  // XXX: Proper name for this?
+  private static final String TEXT_BLOCK_MARKER = "\"\"\"";
+
   private SourceCode() {}
 
+  /**
+   * Tells whether the targeted Java language level supports text blocks.
+   *
+   * @param state A {@link VisitorState} from which the targeted Java language level can be derived.
+   * @return {@code true} iff text block expressions are supported.
+   */
+  // XXX: Add tests!
+  public static boolean isTextBlockSupported(VisitorState state) {
+    // XXX: String comparison is for JDK 11 compatibility. Is there a better way?
+    return Target.instance(state.context).toString().compareTo("JDK1_15") >= 0;
+    // return Target.instance(state.context).compareTo(Target.JDK1_15) >= 0;
+  }
```

Comment:
Note to future self: there's a better way of doing this; see #3646. We can then also revert all references to `com.sun.tools.javac.jvm` in this PR.